src/components/search-bar/search-bar.tsx

```diff
@@ -102,7 +102,7 @@ export const SearchBar = ({
   }
 
   function optionIncludesValueFilter(option: DropDownOption<string>) {
-    const optionVal = option.value;
+    const optionVal = option.value as string;
```

Comment:
I am using the generic to limit the id type even further. 
When this is used as the set of id's
```
const languages = [a, b, c] as const;
```
option.id is typed to be `a  |  b |  c` instead of `string`

But I agree that we should limit the id to extend `string` instead of `React.Key`, I forget we can further limit the value used for the react key. 

Would we want two generics on the `DropDownOption` component?

Reply:
yeah, that makes sense -- two generics to filter down the type.

Reply:
Cool. I just added it